src/pages/topic-page-template.jsx

```diff
@@ -0,0 +1,36 @@
+import { useEffect, useState } from "react";
+import { useParams } from "react-router-dom";
+import { fetchArticlesByTopic } from "../utils/api-calls";
+import Articles from "../components/articles";
+import Divider from "../components/divider";
+import LatestArticlePreview from "../components/latest-article-preview";
+import Header from "../components/header";
+
+export default function TopicPageTemplate() {
+    const [articles, setArticles] = useState([])
+    const [isLoading, setIsLoading] = useState(true)
+    const { topic_slug } = useParams();
+
+    useEffect(() => {
+        fetchArticlesByTopic(topic_slug).then((data) => {
```

Comment:
I think we need to set loading to true at the start of the useEffect here. The useEffect will be invoked twice, once on initial render and once after topic_slug is updated. Hence the fetchArticles will be invoked twice.